### PR TITLE
Workflow for automatic releases added

### PR DESCRIPTION
**Pull request by Ninja243**
This workflow auto-increments the version number, creates a tag for release and creates a GitHub release named after the current version number.

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
+name: Build and release SDK 
+on:
+  workflow_dispatch:
+  pull_request:
```
Comment (aavshr) [inline .github/workflows/release.yml:4]: it should be on pushes to release branch, a merge will be a `push` event

Comment (Ninja243) [reply .github/workflows/release.yml:4]: :facepalm: fixing now

Comment (Ninja243) [reply .github/workflows/release.yml]: Fixed!

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+jobs:
+  build-and-release:
+    runs-on: ubuntu-18.04
+    steps:
+      # Get the code into the container
+      - name: Checkout
+        uses: actions/checkout@v2
+      # Create a new release
+      - name: Semantic version bump
+        uses: rimonhanna/gh-action-bump-version@master
```
Comment (aavshr) [inline .github/workflows/release.yml:16]: this looks like a fork for an action of automatic version bump of npm packages, we should not use `package.json` for golang.

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+          content="${content//$'\r'/'%0D'}"
+          echo "::set-output name=packageJson::$content"
+      - name: GH Release
+        uses: softprops/action-gh-release@v1
+        with:
+          prerelease: false
+          name: "Deta Go ${{fromJson(steps.set_var.outputs.packageJson).version}}"
+          tag_name: ${{fromJson(steps.set_var.outputs.packageJson).version}}
+          draft: false
+          body: 'Deta SDK for Golang. Documentation available at https://docs.deta.sh/docs/base/sdk.'
```
Comment (aavshr) [inline .github/workflows/release.yml:40]: this is the body for the release (change log), should be changes that between releases.

**Review by aavshr — CHANGES_REQUESTED** [review]
thanks for the PR @Ninja243. Made a few comments.

**Pull request closed without merging**